### PR TITLE
Interrupt wait time fix

### PR DESCRIPTION
**Pull request by DyrellC**
The `interruptAttachToTangle` was being called, and then the response for `attachToTangle` was looked for too quickly for the node to return the null response for the `attachToTangle` call. The one second pause gives the response time to return and become present in  `world.responses` for the next step.

### CODE REVIEW

File: PythonRegression/tests/features/steps/api_test_steps.py [hunk 1]
```diff
@@ -98,28 +94,31 @@ def wait_for_step(step,time):
...
 
-        assert response_value == expected_value, \
-            'Returned: {} does not match the expected value: {}'.format(response_value,expected_value)
+        for count in range(len(keys)):
+            response_key = response_keys[count]
+            response_value = response_list[response_key]
+            expected_value = expected_values[response_key]
+
+            assert response_value == expected_value, \
+                'Returned: {} does not match the expected value: {}'.format(response_value,expected_value)
```
[the remark this answers is not in the document]
Comment (DyrellC) [reply PythonRegression/tests/features/steps/api_test_steps.py]: I've already prepared a breakdown of functionality like the response parsing in the refactor that's coming.

File: PythonRegression/util/threading_logic/pool_logic.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+def start_pool(function,iterations,args):
+    iteration_list = [int(iterations)]
+    pool = Pool(processes=4)
+    list_of_threads = {pool.apply_async(function,args): iteration for iteration in iteration_list}
+    logger.info(list_of_threads)
+    return list_of_threads
+
+
+def fetch_results(id, timeout):
```
[the remark this answers is not in the document]
Comment (DyrellC) [reply PythonRegression/util/threading_logic/pool_logic.py]: The `fetch_results` checks the thread that's been supplied to see if the result has returned within the timeout. If it has not, then it logs an exception. If it does have a response, then it returns that response for comparison.

File: PythonRegression/util/threading_logic/pool_logic.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+from multiprocessing.dummy import Pool
+import multiprocessing as mp
+import logging
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+def start_pool(function,iterations,args):
+    #Assigns the process worker pool to either the cpu count or the number of nodes for the call, whichever is lower
+    pool = Pool(processes=len(args) if len(args) <= mp.cpu_count() else mp.cpu_count())
```
Comment (GalRogozinski) [inline PythonRegression/util/threading_logic/pool_logic.py]: Why is the number of "processes" limited by the cpu count?
1. We are talking about networking I/O which is limited by bandwidth not cpu.
2. Regardless of 1, since you are using the `dummy` implementation your "processes" are actually green threads as the GIL doesn't allow real parallelism, so what's the point of the limit? (If we bring (1) back, I think you do get parallelism since you are not utilizing the cpu).

In short you can just remove the limit.

File: PythonRegression/tests/features/steps/api_test_steps.py [hunk 4]
```diff
@@ -69,82 +69,106 @@ def threaded_call(step,apiCall,node):
...
+            del response_list['duration']
+        response_keys = response_list.keys()
 
-    #Confirm that the lists are of equal length before comparing
-    assert len(keys) == len(response_keys), 'Response: {} does not contain the same number of arguments: {}'.format(keys,response_keys)
+        '''
+        Prepares gherkin table based key dictionary for comparison with response values. 
+        Takes the key and value from the table and places it in a dictionary with the appropriate value type. 
+        Further on this above.
+        '''
```
[the remark this answers is not in the document]
Comment (DyrellC) [reply PythonRegression/tests/features/steps/api_test_steps.py]: In the documentation I've seen it's stated that single or multi-line docs should be in triple quotations wherever they are.

File: PythonRegression/util/threading_logic/pool_logic.py [hunk 5]
```diff
@@ -6,7 +6,7 @@
 
 def start_pool(function,iterations,args):
     #Assigns the process worker pool to either the cpu count or the number of nodes for the call, whichever is lower
```
Comment (GalRogozinski) [inline PythonRegression/util/threading_logic/pool_logic.py]: you forgot to update the comment.
By the way you can change it to a doc string since it right below the function definition

File: PythonRegression/tests/features/steps/api_test_steps.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply PythonRegression/tests/features/steps/api_test_steps.py]: Nice docstring, but it should be *inside the definition* of `api_utils.prepare_options`.
Think that you will call this function many times in other parts of the code. Your IDE should be able to display this docstring from wherever you see it.

Look at the examples in the link I gave you again.

File: PythonRegression/tests/features/steps/api_test_steps.py [hunk 6]
```diff
@@ -38,7 +38,7 @@
         responseList:   Same as responseValue, ecept it places the results into a list
         bool:           Bool argument, returns True or False
  
-'''
+"""
```
Comment (GalRogozinski) [inline PythonRegression/tests/features/steps/api_test_steps.py]: hmm
question, shouldn't this docstring be under the method as described in PEP 257?

File: PythonRegression/tests/features/steps/api_test_steps.py [hunk 7]
```diff
@@ -91,35 +91,23 @@ def wait_for_step(step,time):
...
-    #Prepare response list for comparison
+    """Prepare response list for comparison"""
     logger.debug(world.responses)
     future_results = world.config['future_results'][apiCall]
 
     for result in future_results:
         response_list = pool.fetch_results(result,1)
 
-        #Exclude duration from response list
+        '''Exclude duration from response list'''
```
Comment (GalRogozinski) [inline PythonRegression/tests/features/steps/api_test_steps.py]: Another question:
replaceing `#` -> with `'''`, why did you find that this is needed. Didn't see anything about in PEP 257...

Comment (DyrellC) [reply PythonRegression/tests/features/steps/api_test_steps.py]: IntelliJ was suggesting I use triple quotes, I wasn't really sure why but I thought i'd switch it over anyways, i'll just switch it back.
